go/apps/mixnet/README.md

```diff
@@ -0,0 +1,89 @@
+TPM; finally, the mixnet code itself is measured and generates a private/
+public key pair, which is attested to by the OS. Hence, trust in the mixnet
+routers to faithfully carry out the protocol is reduced to correct
+provisioning of the policy key.
+
+We consider a global passive adversary who observes all communications between
+senders and mixnet routers, mixnet routers and other mixnet routers, and mixnet
+routers and recipients. The adversary may also send messages on channels she
+observes. Since the service is anonymous, the adversary is allowed to control
+any number of senders or recipients. The effect is that the state of that peer
+is exposed, including any and all cryptographic keys. We assume that the policy
+key was correctly provisioned; if the code implements the protocol correctly and
+is properly isolated during its execution, we claim this precludes the
+possibility of exposing a router’s state. (This is a strong claim offered here
+without proof. CloudProxy provides assurance that the expected program is 
+running; assuming the code is "correct", a formal treatment of our protocol 
+should reduce the adversary's control of the routers to standard cryptographic 
+assumptions: in particular, CDH on elliptic curves.)
+
+The intended property of communications over the mixnet is unlinkability in the
+sense of [1]. Consider a set of senders _S_ where _|S| = n_ and a set of 
+recipients _T_. Each sender chooses one recipient as well as a message to send 
+so that _M : S→T_ is an onto mapping. The messages are transmitted to their 
+respective recipients over the mixnet; the adversary succeeds if it outputs 
+_(s, t)_ such that _M(s) = t_, unless it controls both _s_ and _t_. We say that 
+communication over the mixnet is _unlinkable_ if for any adversary the 
+probability of success is less than _1/n_ plus some negligible value.
+
+_Alternative definition:_ as above, except the adversary chooses the messages to
+be sent. This change may make it easier to analyze the unlinkability of a
+particular protocol. However, this would require the recipients to participate
+in the protocol, since messages exiting the mixnet must be encrypted. 
+
+For a protocol to achieve security in this sense, the messages must all have the 
+same length; of course, this is not always reasonable in practice. Mixnets 
+address this by splitting messages into fixed-length cells. Senders split 
+messages into cells (zero-padding the last cell as needed) and send them to the 
+first router where they are added to a queue. At each round the router waits 
+until there are _m_ cells in the queue from _m_ distinct senders and transmits 
+these simultaneously to achieve anonymity. 
+
+Extending the definition of unlinkability to a mixnet that divides messages into
+cells and transmits at rounds is indeed challenging, as varying message lengths 
+exposes traffic patterns to the adversary. One way to mitigate this problem is 
```

Comment:
"varying message lengths" is ambiguous. Maybe "challenging: the presence of variable-length messages exposes traffic patterns to the adversary"